scripts/package/preinstall.sh

```diff
@@ -16,18 +16,70 @@
 
 set -e
 
-username="observiq-otel-collector"
+username="bdot"
+legacy_username="observiq-otel-collector"
 
-if getent group "$username" > /dev/null 2>&1; then
-    echo "Group ${username} already exists."
-else
-    groupadd "$username"
-fi
+# Install creates the user and group for the collector
+# service. This function is idempotent and safe to call
+# multiple times.
+install() {
+    # Return early without output if the user and group already exist.
+    # This will help avoid confusion with the output in migrate_user().
+    if id "$username" > /dev/null 2>&1 && getent group "$username" > /dev/null 2>&1; then
+        return
+    fi
 
-if id "$username" > /dev/null 2>&1; then
-    echo "User ${username} already exists"
-    exit 0
-else
-    useradd --shell /sbin/nologin --system "$username" -g "$username"
-fi
+    if getent group "$username" > /dev/null 2>&1; then
+        echo "Group ${username} already exists."
+    else
+        groupadd "$username"
+    fi
 
+    if id "$username" > /dev/null 2>&1; then
+        echo "User ${username} already exists"
+        exit 0
+    else
+        useradd --shell /sbin/nologin --system "$username" -g "$username"
+    fi
+}
+
+# migrate_user migrates the legacy user to the new username.
+migrate_user() {
+    _migrate_user
+    _migrate_group
+
+    echo "User migration complete."
+}
+
+_migrate_user() {
+    if ! id "$legacy_username" > /dev/null 2>&1; then
+        echo "Skipping user migration: Legacy user ${legacy_username} does not exist."
+        return
+    fi
+
+    if id "$username" > /dev/null 2>&1; then
+        echo "Skipping user migration: User ${username} already exists."
+        return
+    fi
+
+    echo "Renaming user ${legacy_username} to ${username}"
+    usermod -l "$username" "$legacy_username"
+}
+
+_migrate_group() {
+    if ! getent group "$legacy_username" > /dev/null 2>&1; then
+        echo "Skipping group migration: Legacy group ${legacy_username} does not exist."
+        return
+    fi
+
+    if getent group "$username" > /dev/null 2>&1; then
+        echo "Skipping group migration: Group ${username} already exists."
+        return
+    fi
+
+    echo "Renaming group ${legacy_username} to ${username}"
+    groupmod -n "$username" "$legacy_username"
```

Comment:
This won't work for AIX (it doesn't have `groupmod`, despite having `usermod`) - I believe I have an alternative I can put in once I get to pulling these changes into the AIX branch. Just to remind me, could we put a TODO comment here?